client/my-sites/controller.js

```diff
@@ -87,9 +89,12 @@ function createNavigation( context ) {
 		basePath = sectionify( context.pathname );
 	}
 
+	const SidebarComponent = config.isEnabled( 'jetpack-cloud' ) ? JetpackCloudSidebar : Sidebar;
+
 	return (
 		<NavigationComponent
 			path={ context.path }
+			sidebar={ <SidebarComponent path={ context.path } siteBasePath={ basePath } /> }
```

Comment:
Again, this is a bit hard to follow. Not a blocker, but a more classic approach might be better.

Reply:
Can you explain the classic approach?

Reply:
Here is a good explanation of the pattern used https://www.robinwieruch.de/react-pass-props-to-component#how-to-pass-components-as-props